client/lib/protect-form/index.jsx

```diff
@@ -57,6 +67,41 @@ export const protectForm = WrappedComponent => {
 	};
 };
 
+/*
+ * Declarative variant that takes a 'isChanged' prop.
+ */
+export class ProtectForm extends Component {
+	static propTypes = {
+		isChanged: PropTypes.bool.isRequired
+	}
+
+	// This function binding makes the listener attached to window unique for each
+	// component instance.
+	warnIfChanged = warnIfChanged.bind();
```

Comment:
@spen there are no memory concerns as long as we don't end up with loops that create these components. we mainly want to avoid `bind` inside of React props because those get recreated on every render. also it's not limited to `bind` but arrow functions and closures of any kind.